android-components/components/feature/readerview/src/main/assets/extensions/readerview/readerview-background.js

```diff
@@ -5,16 +5,17 @@
 // This background script is needed to update the current tab
 // and activate reader view.
 
-browser.runtime.onMessage.addListener(async (message, sender, sendResponse) => {
+browser.runtime.onMessage.addListener(message => {
   switch (message.action) {
      case 'addSerializedDoc':
         browser.storage.session.set({ [message.id]: message.doc });
-        break;
+        return Promise.resolve();
      case 'getSerializedDoc':
-       let doc = await browser.storage.session.get(message.id);
-       browser.storage.session.remove(message.id);
-       sendResponse(doc);
-       break;
+        return (async () => {
+          let doc = await browser.storage.session.get(message.id);
+          browser.storage.session.remove(message.id);
+          return doc[message.id];
```

Comment:
@boek this part of the change was necessary to retrieve the actual value we are looking for in the session storage, the API call will be returning an object with the value we are looking for set as the value of the `message.id` key set on that object.